### ISSUE TITLE
COL_DTYPES for regionprops_table differ from regionprops types

### ISSUE THREAD

**Issue by dmyung** [scikit-image/scikit-image#4280]
## Description
In trying the `0.16.2`  `regionprops_table` to extract object measurements, we noticed that some properties we expected float types were coming back as integers.

https://github.com/scikit-image/scikit-image/blob/master/skimage/measure/_regionprops.py#L71

Seeing how the regionprops_table dictionary series are being instantiated, there are conversions happening here when creating the dataframe-ready object, that we did not encounter in the `regionprops` iterator method.

## Way to reproduce
```python
from skimage.data import binary_blobs
from skimage.measure import regionprops, regionprops_table
import numpy as np


blobs = binary_blobs().astype(np.uint8)

rprops = regionprops(blobs)
print("regionprops iter result")
print(rprops[0].centroid)
print(rprops[0].local_centroid)


blobs_table = regionprops_table(blobs, properties=['centroid', 'local_centroid'])
print("regionprops table result")
print(blobs_table)


### Output

> python bblob.py 
regionprops iter result
(258.17323303222656, 248.64542388916016)
(258.17323303222656, 248.64542388916016)
regionprops table result
{'centroid-0': array([258]), 'centroid-1': array([248]), 'local_centroid-0': array([258]), 'local_centroid-1': array([248])}

```

## Version information
```python
3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 21:52:21) 
[GCC 7.3.0]
Linux-4.15.0-66-generic-x86_64-with-debian-buster-sid
scikit-image version: 0.16.2
numpy version: 1.16.4
```

**Comment by alexdesiqueira** [comment]
Hi @dmyung,
you're right: `centroid` is being returned as an integer, as seen in `COL_DTYPES = {..., 'centroid': int, ...}`.
I wrote and rewrote this sentence trying to defend that; I am thinking on using the pixel coordinates on the image (should be `int`), but I guess @jni will do a better job than me. :blush:

**Comment by jni** [comment]
Nope, it is indefensible! Simply a mistake that made it through review. The dtype should be changed to float, ideally a test should be made that for every property, the dtype is the same between `regionprops` and `regionprops_table`, and then this should be backported to v0.16.x for the next point release.

**Comment by jni** [comment]
Thank you @dmyung for reporting this back to us! I hope we'll have a bug fix out soon!

**Comment by rfezzani** [comment]
Isn't this issue fixed by  #4311?

**Comment by jni** [comment]
Yes indeed! Looks like we missed the keyword "fixes" or "closes" there. Oops! Thanks for the triage, @rfezzani!
